pygmt/src/which.py

```diff
@@ -56,22 +50,25 @@ def which(fname, **kwargs):
 
     Returns
     -------
-    path : str or list
+    path
         The path(s) to the file(s), depending on the parameters used.
 
     Raises
     ------
     FileNotFoundError
         If the file is not found.
     """
-    with GMTTempFile() as tmpfile:
-        with Session() as lib:
+    with Session() as lib:
+        with lib.virtualfile_out(kind="dataset") as vouttbl:
             lib.call_module(
                 module="which",
-                args=build_arg_string(kwargs, infile=fname, outfile=tmpfile.name),
+                args=build_arg_string(kwargs, infile=fname, outfile=vouttbl),
             )
-        path = tmpfile.read().strip()
-    if not path:
+            result = lib.virtualfile_to_dataset(vfname=vouttbl, output_type="pandas")
```

Comment:
We can set output_type to `numpy` to simplify the code below no?

Reply:
Yes, it's not very efficient. Instead, we can add another internal type `output_type="string"`, which returns the vector of the trailing text only. It's useful if we know that a module only outputs text strings like `which`. Actually, GMT provides an API function [`GMT_Get_Strings`](https://docs.generic-mapping-tools.org/dev/devdocs/api.html#gmt-get-strings) which does exactly the same thing (if I understand it correctly). We can also wrap that API function instead.

Reply:
> Instead, we can add another internal type `output_type="string"`, which returns the vector of the trailing text only. It's useful if we know that a module only outputs text strings like `which`.

Done in PR #3157.

> Actually, GMT provides an API function [`GMT_Get_Strings`](https://docs.generic-mapping-tools.org/dev/devdocs/api.html#gmt-get-strings) which does exactly the same thing (if I understand it correctly). We can also wrap that API function instead.

This API function requires the data family to be `GMT_IS_VECTOR`/`GMT_IS_MATRIX`, so it cannot be used in this case (family is `GMT_IS_DATASET`). xref: https://github.com/GenericMappingTools/gmt/blob/fcb795ef196714fe51f7e5e68d30a18e981294d0/src/gmt_api.c#L15785